### PR TITLE
Disallow case change in bin/content-push

### PR DESCRIPTION
**Pull request by davidsbailey**
```
Davids-MacBook-Pro-2:~/src/cdo2 (disallow-case-change)$ git mv --force pegasus/sites.v3/code.org/public/images/hopscotch.png pegasus/sites.v3/code.org/public/images/HOPSCOTCH.png
Davids-MacBook-Pro-2:~/src/cdo2 (disallow-case-change)$ bin/content-push 
Who are you? dave
Hi dave!

Aborting due to case-only rename: R  pegasus/sites.v3/code.org/public/images/hopscotch.png -> pegasus/sites.v3/code.org/public/images/HOPSCOTCH.png
To fix this, run the following commands:

git mv pegasus/sites.v3/code.org/public/images/HOPSCOTCH.png pegasus/sites.v3/code.org/public/images/HOPSCOTCH.png.tmp

git commit --only pegasus/sites.v3/code.org/public/images/hopscotch.png pegasus/sites.v3/code.org/public/images/HOPSCOTCH.png.tmp -m 'temporarily rename pegasus/sites.v3/code.org/public/images/hopscotch.png -> pegasus/sites.v3/code.org/public/images/HOPSCOTCH.png.tmp'

git mv pegasus/sites.v3/code.org/public/images/HOPSCOTCH.png.tmp pegasus/sites.v3/code.org/public/images/HOPSCOTCH.png

bin/content-push
```

### CODE REVIEW

File: bin/content-push [hunk 1]
```diff
@@ -14,12 +14,28 @@ def ask_for_name
   name
 end
 
+def disallow_case_change
+  `git status --porcelain --untracked-files=all #{CONTENT_PATHS}`.split("\n").each do |line|
+    if (%r{^R  ([^ ]+) -> ([^ ]+)}.match(line) || %r{^R  ("[^"]+") -> ("[^"]+")}.match(line)) && $1.casecmp($2) == 0
```
Comment (aoby) [inline bin/content-push:19]: Are there always the same number of spaces? Safe to use this over whitespace: `\s+`?

**Comment by aoby** [comment]
lgtm. Thanks for addressing this. It's been a headache recently.

File: bin/content-push (hunk 1 above)
Comment (davidsbailey) [inline bin/content-push:19]: Good point. Upon further investigation, it appears this format is guaranteed to never change according to `git help status`:

> The porcelain format is similar to the short format, but is guaranteed not to change in a
> backwards-incompatible way between Git versions or based on user configuration. This makes it ideal
> for parsing by scripts.

**Comment by joshlory** [comment]
This is super cool!  Is it possible to do this in the pre-commit hook as well, for folks who don't use the `bin/content-push` script?

**Comment by davidsbailey** [comment]
yes, here: https://github.com/code-dot-org/code-dot-org/blob/staging/tools/hooks/pre-commit

**Pull request merged**